tests/test_modeling_realm.py

```diff
@@ -0,0 +1,565 @@
+            model = model_class(config)
+            model.to(torch_device)
+            model.train()
+            inputs = self._prepare_for_class(inputs_dict, model_class, return_labels=True)
+            loss = model(**inputs).loss
+            loss.backward()
+
+    @slow
+    def test_embedder_from_pretrained(self):
+        model = RealmEmbedder.from_pretrained("qqaatw/realm-cc-news-pretrained-embedder")
+        self.assertIsNotNone(model)
+
+    @slow
+    def test_encoder_from_pretrained(self):
+        model = RealmKnowledgeAugEncoder.from_pretrained("qqaatw/realm-cc-news-pretrained-bert")
+        self.assertIsNotNone(model)
+
+    @slow
+    def test_open_qa_from_pretrained(self):
+        # TODO: TF record dataset
+        model = RealmForOpenQA.from_pretrained(
+            "qqaatw/realm-orqa-nq-searcher", "qqaatw/realm-orqa-nq-reader", BLOCK_RECORDS_PATH
+        )
+        self.assertIsNotNone(model)
+
+    @slow
+    def test_reader_from_pretrained(self):
+        model = RealmReader.from_pretrained("qqaatw/realm-orqa-nq-reader")
+        self.assertIsNotNone(model)
+
+    @slow
+    def test_retriever_from_pretrained(self):
+        model = RealmScorer.from_pretrained("qqaatw/realm-cc-news-pretrained-retriever")
+        self.assertIsNotNone(model)
+
+    @slow
+    def test_searcher_from_pretrained(self):
+        # TODO: TF record dataset
+        model = RealmSearcher.from_pretrained("qqaatw/realm-orqa-nq-searcher", BLOCK_RECORDS_PATH)
+        self.assertIsNotNone(model)
+
+
+@require_torch
+class RealmModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_embedder(self):
+        retriever_projected_size = 128
+
+        model = RealmEmbedder.from_pretrained("qqaatw/realm-cc-news-pretrained-embedder")
+        input_ids = torch.tensor([[0, 1, 2, 3, 4, 5]])
+        output = model(input_ids)[0]
+
+        expected_shape = torch.Size((1, retriever_projected_size))
+        self.assertEqual(output.shape, expected_shape)
+
+        expected_slice = torch.tensor([[-0.0714, -0.0837, -0.1314]])
+        self.assertTrue(torch.allclose(output[:, :3], expected_slice, atol=1e-4))
+
+    @slow
+    def test_inference_encoder(self):
+        num_candidates = 2
+        vocab_size = 30522
+
+        model = RealmKnowledgeAugEncoder.from_pretrained(
+            "qqaatw/realm-cc-news-pretrained-bert", num_candidates=num_candidates
+        )
+        input_ids = torch.tensor([[0, 1, 2, 3, 4, 5], [6, 7, 8, 9, 10, 11]])
+        relevance_score = torch.tensor([[0.3, 0.7]], dtype=torch.float32)
+        output = model(input_ids, relevance_score=relevance_score)[0]
+
+        expected_shape = torch.Size((2, 6, vocab_size))
+        self.assertEqual(output.shape, expected_shape)
+
+        expected_slice = torch.tensor([[[-11.0888, -11.2544], [-10.2170, -10.3874]]])
+
+        self.assertTrue(torch.allclose(output[1, :2, :2], expected_slice, atol=1e-4))
+
+    @slow
+    def test_inference_open_qa(self):
```

Comment:
In a first step, let's just make sure at every iteration that this test still passes. Later we can make all the other tests work.